scripts/build/mkversioninfo

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env sh
+set -eu
+
+: "${VERSION=}"
+: "${GITCOMMIT=}"
+: "${TARGET=}"
+
+# Create version quad for Windows of the form major.minor.patch.build
+VERSION_QUAD=$(printf "%s" "$VERSION" | sed -re 's/^([0-9.]*).*$/\1/' | sed -re 's/\.$//' | sed -re 's/^[0-9]+$/\0\.0/' | sed -re 's/^[0-9]+\.[0-9]+$/\0\.0/' | sed -re 's/^[0-9]+\.[0-9]+\.[0-9]+$/\0\.0/')
+
+# Generate versioninfo.json to be able to create a syso file which contains
+# Microsoft Windows Version Information and an icon using goversioninfo.
+# Example: https://github.com/josephspurrier/goversioninfo/blob/master/testdata/resource/versioninfo.json
+cat > ./cmd/docker/versioninfo.json <<EOL
+{
+  "FixedFileInfo":
+  {
+    "FileVersion": {
+      "Major": $(echo "$VERSION_QUAD" | cut -d. -f1),
+      "Minor": $(echo "$VERSION_QUAD" | cut -d. -f2),
+      "Patch": $(echo "$VERSION_QUAD" | cut -d. -f3),
+      "Build": $(echo "$VERSION_QUAD" | cut -d. -f4)
+    },
+    "FileFlagsMask": "3f",
+    "FileFlags ": "00",
+    "FileOS": "040004",
+    "FileType": "01",
+    "FileSubType": "00"
+  },
+  "StringFileInfo":
+  {
+    "Comments": "",
+    "CompanyName": "Docker, Inc.",
+    "FileDescription": "Docker Client",
+    "FileVersion": "${VERSION_QUAD}",
+    "InternalName": "",
+    "LegalCopyright": "Copyright (c) 2013-2021 Docker Inc. All rights reserved.",
```

Comment:
Thinking if we should keep this empty (for now). Reading https://docs.microsoft.com/en-us/windows/win32/menurc/stringfileinfo-block, this field should not just contain a single copyright, but _all_ noticed etc (which in this case may require all noticed, also from dependencies etc to be included). Given that it's _optional_, perhaps we should avoid that (for now); probably better to "not" include it that to include _incorrect_ / _partial_ info;

```
LegalCopyright
Copyright notices that apply to the file. This should include the full text of
all notices, legal symbols, copyright dates, and so on. This string is optional.
```

Reply:
Not an legal expert but this disclaimer is not exactly right (cc @justincormack @chris-crone). Some various bits around the industry:

### Slack

![image](https://user-images.githubusercontent.com/1951866/134917868-7790eec4-27b2-4de0-9d7e-0aa94471e9e1.png)

### Discord

![image](https://user-images.githubusercontent.com/1951866/134917757-6e41a1ff-55d0-44db-bd5e-b0ca7e04fef8.png)

### Firefox

![image](https://user-images.githubusercontent.com/1951866/134917966-170e085f-f8b0-417a-a921-3586c29ecd1a.png)

### Chrome

![image](https://user-images.githubusercontent.com/1951866/134918069-ea561e22-f295-43fa-a395-d0cc15f80dd9.png)

Reply:
Changed to have the same as Docker Desktop:

![image](https://user-images.githubusercontent.com/1951866/135863767-3e79d85c-c1e1-4b6b-8c43-ee3843a4bd3e.png)